### PR TITLE
update print dialog ui and behaviour

### PR DESCRIPTION
**Pull request by myarjunar**
### What does it fix?
* Ticket: #4705 
* Funded by: DFAT
* Description: 
impact report - preview _159

### Checklist:
- [ ] Commit messages are descriptive and explain the rationale for changes
- [ ] Add to the changelog in metadata.txt if it's a new feature
- [ ] Unit test for new code added
- [ ] Request someone to review or test your PR

### CODE REVIEW

File: safe/gui/tools/print_report_dialog.py [hunk 1]
```diff
@@ -65,36 +72,96 @@ def __init__(self, impact_function, iface, dock=None, parent=None):
...
+                        override_template_found = filename
+
+        if override_template_found:
+            string_format = tr('*Template override found: {template_path}')
+            self.override_template_found_label.setText(
+                string_format.format(template_path=override_template_found))
+
         # additional buttons
-        self.button_print_pdf = QtGui.QPushButton(self.tr('Print as PDF'))
+        self.button_print_pdf = QtGui.QPushButton(self.tr('Open as PDF'))
```
Comment (ismailsunni) [inline safe/gui/tools/print_report_dialog.py]: Avoid to use `self.tr`, just use `tr` to reduce the number of translation. Use context if you want to make it translated differently.

File: safe/gui/tools/print_report_dialog.py [hunk 2]
```diff
@@ -65,36 +72,96 @@ def __init__(self, impact_function, iface, dock=None, parent=None):
...
         self.button_print_pdf.setObjectName('button_print_pdf')
```
Comment (ismailsunni) [inline safe/gui/tools/print_report_dialog.py]: Why do you need to set the object name?

File: safe/gui/tools/print_report_dialog.py [hunk 3]
```diff
@@ -65,36 +72,96 @@ def __init__(self, impact_function, iface, dock=None, parent=None):
...
         self.button_print_pdf.setToolTip(self.tr(
```
Comment (ismailsunni) [inline safe/gui/tools/print_report_dialog.py]: Use `tr`

File: safe/gui/tools/print_report_dialog.py [hunk 4]
```diff
@@ -376,32 +435,110 @@ def open_in_composer(self):
...
                 return
 
-    def accept(self):
-        """Method invoked when OK button is clicked."""
+    def prepare_components(self):
+        """Prepare components that are going to be generated based on
+        user options.
 
-        self.save_state()
+        :return: Updated list of components
```
Comment (ismailsunni) [inline safe/gui/tools/print_report_dialog.py]: Dot in the end of string.

File: safe/gui/tools/print_report_dialog.py [hunk 5]
```diff
@@ -376,32 +435,110 @@ def open_in_composer(self):
...
                 self.template_combo.currentIndex())
-        else:
-            self.selected_template_path = self.template_path.text()
+        elif self.search_on_disk_radio.isChecked():
+            selected_template_path = self.template_path.text()
             if not exists(self.selected_template_path):
                 # noinspection PyCallByClass,PyTypeChecker
                 QtGui.QMessageBox.warning(
                     self,
                     self.tr('InaSAFE'),
```
Comment (ismailsunni) [inline safe/gui/tools/print_report_dialog.py]: `tr`

File: safe/gui/tools/print_report_dialog.py [hunk 6]
```diff
@@ -376,32 +435,110 @@ def open_in_composer(self):
...
-                    self.tr('Please select a valid template before printing. '
-                            'The template you choose does not exist.'))
+                    self.tr(
```
Comment (ismailsunni) [inline safe/gui/tools/print_report_dialog.py]: Use `tr`

File: safe/gui/tools/print_report_dialog.py [hunk 7]
```diff
@@ -439,15 +576,24 @@ def template_chooser_clicked(self):
             self.tr('QGIS composer templates (*.qpt *.QPT)'))
         self.template_path.setText(file_name)
 
-    def toggle_template_selectors(self, checked):
-        if checked:
+    def toggle_template_selector(self):
```
Comment (ismailsunni) [inline safe/gui/tools/print_report_dialog.py]: Docstring

File: safe/gui/tools/print_report_dialog.py (hunk 2 above)
Comment (myarjunar) [reply safe/gui/tools/print_report_dialog.py]: I'm using it as a sender indentifier 
```
            if sender_name == 'button_print_pdf':
                self.create_pdf = True
                self.open_as_pdf()
            else:
                self.create_pdf = False
                self.open_in_composer()
```

Comment (ismailsunni) [reply safe/gui/tools/print_report_dialog.py]: Ah I see... Another approach would make two different method, and connect them to the buttons. But I think it's also fine, new trick for me :D

Comment (myarjunar) [reply safe/gui/tools/print_report_dialog.py]: lazy man approach :joy:

File: safe/gui/tools/print_report_dialog.py [hunk 8]
```diff
@@ -434,20 +571,37 @@ def template_chooser_clicked(self):
...
+
+        if self.search_on_disk_radio.isChecked():
             self.template_path.setEnabled(True)
             self.template_chooser.setEnabled(True)
+        else:
+            self.template_path.setEnabled(False)
+            self.template_chooser.setEnabled(False)
+
+    def toggle_no_map_radio(self, checked):
+        """Auto-connect slot when no map radio is clicked.
```
Comment (Gustry) [inline safe/gui/tools/print_report_dialog.py]: Small note, these are not auto-connect ;-) (same as line 582)

File: safe/gui/tools/print_report_dialog.py [hunk 9]
```diff
@@ -434,20 +571,37 @@ def template_chooser_clicked(self):
...
+
+        .. versionadded: 4.3.0
+        """
+        if checked:
+            self.button_open_composer.setEnabled(False)
+        else:
+            self.button_open_composer.setEnabled(True)
```
Comment (Gustry) [inline safe/gui/tools/print_report_dialog.py]: I often write this too and then I noticed an easier way very late in my coding life ;-)
`self.button_open_composer.setEnabled(not checked)`

Comment (myarjunar) [reply safe/gui/tools/print_report_dialog.py]: ha!

Comment (ismailsunni) [reply safe/gui/tools/print_report_dialog.py]: yeah, me too @Gustry

File: safe/gui/tools/print_report_dialog.py [hunk 10]
```diff
@@ -590,10 +594,11 @@ def toggle_template_selector(self):
...
     def toggle_no_map_radio(self, checked):
-        if checked:
-            self.button_open_composer.setEnabled(False)
-        else:
-            self.button_open_composer.setEnabled(True)
+        """Slot when no map radio is clicked.
+
+        .. versionadded: 4.3.0
+        """
+        self.button_open_composer.setEnabled(not checked)
```
Comment (Gustry) [inline safe/gui/tools/print_report_dialog.py]: So now, you can remove this slot and connect the signal to `self.button_open_composer.setDisabled` ;-)

**Comment by timlinux** [comment]
@myarjunar can you address the few comments then merge?

**Pull request merged**